static/app/utils/replays/hooks/useIsLive.tsx

```diff
@@ -0,0 +1,24 @@
+import {useState} from 'react';
+
+import type ReplayReader from 'sentry/utils/replays/replayReader';
+
+type Props = {
+  replayReader: ReplayReader | null;
+};
+
+function useIsLive({replayReader}: Props) {
+  const [isLive, setIsLive] = useState<boolean>(replayReader?.getIsLive() ?? false);
+
+  if (isLive && replayReader) {
+    setTimeout(
+      () => {
+        setIsLive(false);
+      },
+      replayReader.getStartTimestampMs() + 60 * 60 * 1000 - Date.now()
+    );
```

Comment:
The setTimeout should be inside a `useEffect` otherwise i'll be re-created each time this renders (or each time the parent renders)

Also, that useEffect should probably return a callback that calls`clearTimeout` when this gets unmounted.
Actually, we have a helper `useTimeout` that would help handle all the cases